src/commands/CaptainCommand.ts

```diff
@@ -75,155 +73,33 @@ export default class CaptainCommand implements Command {
       return;
     }
 
-    const player = PlayerData.playerDataList.find(
-      (p) => p.getDiscordUserId() === user.id
-    );
-    if (!player) {
+    if (!GameManager.getGame().announced) {
       await interaction.reply({
-        content: "This player hasn't registered!",
+        content: "No game has been announced yet!",
         ephemeral: true,
       });
       return;
     }
 
-    const inGameName = player.getInGameName();
-    if (!GameData.getPlayers().includes(inGameName)) {
+    const player = GameManager.getGame()
+      .getPlayers()
+      .find((player) => player.discordSnowflake === user.id);
+
+    if (!player) {
       await interaction.reply({
-        content: `${inGameName} is not in the current game players list. Captain cannot be set.`,
+        content: "This player hasn't registered!",
         ephemeral: true,
       });
       return;
     }
 
-    console.log(
-      `Setting captain for ${teamColor} team. Checking current captain...`
-    );
-    const currentTeamCaptain = getCaptainByTeam(teamColor as "blue" | "red");
-    console.log(
-      `Current ${teamColor} team captain: ${currentTeamCaptain?.getDiscordUserId() || "None"}`
-    );
-
-    if (currentTeamCaptain) {
-      const currentCaptainMember = interaction.guild.members.cache.get(
-        currentTeamCaptain.getDiscordUserId()
-      );
-      if (currentCaptainMember) {
-        console.log(
-          `Removing captain and team role from current captain: ${currentCaptainMember.user.username}`
-        );
-        await currentCaptainMember.roles
-          .remove(captainRoleId)
-          .catch((error) =>
-            console.error(
-              `Failed to remove captain role from current team captain: ${error}`
-            )
-          );
-
-        const teamRoleId =
-          teamColor === "blue" ? blueTeamRoleId : redTeamRoleId;
-        await currentCaptainMember.roles
-          .remove(teamRoleId)
-          .catch((error) =>
-            console.error(
-              `Failed to remove team role from current team captain: ${error}`
-            )
-          );
-
-        currentTeamCaptain.setCaptain(false);
-      }
-    }
-
-    const oppositeTeamColor = teamColor === "blue" ? "red" : "blue";
-    const oppositeCaptain = getCaptainByTeam(
-      oppositeTeamColor as "blue" | "red"
```

Comment:
why are we removing all these safety checks to verify the player is being set to the right team as a captain, unselecting previous captains, preventing duplicates, etc?

Reply:
added back in few latest commits